src/components/Legend.tsx

```diff
@@ -160,8 +162,21 @@ export function Legend({
         if (value.range) {
           return (
             <g
+              onClick={(event) => {
+                onClick?.({ event, id: value.id });
+                if (showHide) {
+                  legendDispatch({
+                    type: 'ADD_LEGEND_LABEL',
+                    payload: {
+                      ...value,
+                      visibility: !value.visibility,
+                    },
```

Comment:
Please add a new action type called `TOGGLE_VISIBILITY`, which only takes the series id and the reducer can just update the boolean.
This part would be changed to : `legendDispatch({ type: 'TOGGLE_VISIBILITY', payload: { seriesId: value.id } })`